user-interface/src/main/java/life/qbic/datamanager/views/DataManagerLayout.java

```diff
@@ -18,9 +19,9 @@
 public class DataManagerLayout extends AppLayout implements RouterLayout {
 
   private final Div contentArea;
-  private final FooterComponent footerComponent = new FooterComponent();
 
-  protected DataManagerLayout() {
+  protected DataManagerLayout(FooterComponent footerComponent) {
+    Objects.requireNonNull(footerComponent);
```

Comment:
What's the advantage of wiring the footer component in each inheriting layout instead of creating it once in the outmost parent layout?